presto-main/src/main/java/io/prestosql/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -364,10 +369,15 @@ public Expression visitValues(ValuesNode node, Void context)
 
                 List<Object> values = builder.build();
 
-                Domain domain = Domain.none(type);
-
-                if (!values.isEmpty()) {
-                    domain = domain.union(Domain.multipleValues(type, values));
+                Domain domain;
+                if (values.isEmpty()) {
```

Comment:
Do we need to handle `empty` as special case? Should `Domain.multipleValues` handle that 
in order to reduce `iffs`?
For example:
```
Domain.create(ValueSet.copyOf(type, ImmutableList.of()), false).isNone() == true
```
Yet `multipleValues` is implemented as
```
    public static Domain multipleValues(Type type, List<?> values)
    {
        if (values.isEmpty()) {
            throw new IllegalArgumentException("values cannot be empty");
        }
        if (values.size() == 1) {
            return singleValue(type, values.get(0));
        }
        return new Domain(ValueSet.of(type, values.get(0), values.subList(1, values.size()).toArray()), false);
    }
```

Reply:
i do not have a strong opinion. There are exactly 3 non-test call sites for `multipleValues`, so updating it in a follow up would not be a problem.